src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -840,10 +841,21 @@ def save_as_table(
         child: SnowflakePlan,
         source_plan: Optional[LogicalPlan],
         use_scoped_temp_objects: bool,
-        is_generated: bool,  # true if the table is generated internally
-        child_attributes: List[Attribute],
+        creation_source: TableCreationSource,
```

Comment:
add a comment for the creation_source, and why we need that

Reply:
added